### DISCUSSION TITLE
SQLAlchemy 2.0 Declarative data class as FastApi response model - how to exclude field from that response?

### DISCUSSION THREAD

**Discussion by GarikFirst** [fastapi/fastapi#6032] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
// Models
class Base(DeclarativeBase, MappedAsDataclass):
    pass

class Project(Base):
    id: Mapped[int] = mapped_column(init=False, primary_key=True)
    name: Mapped[str] = mapped_column(String())
    …


// Fast API method
@router.post("/", status_code=status.HTTP_201_CREATED)
async def create_project(
    project: Project,
    session: AsyncSession = Depends(get_session),
) -> Project:
    session.add(project)
    return project

// Swagger hint
{
  "id": 0,
  "name": "string",
  ...
}
```

### Description

I use SQL Alchemy 2.0 Declarative data class to declare my database models (see Models in Example Code), and I like to use this model as request body model in Fast API post method  (see Fast API method in Example Code).

And, of course, id field is required for this request and it doesn’t pass validation.

Is there any way to somehow omit id just for request body model (without creating unnecessary Pydantic model just for this method, using just my sqlalchemy model)? 

### Operating System

Linux

### FastAPI Version

0.92.0

### Python Version

3.10.10

**Comment by mr-st0rm** [comment, marked as answer]
no u can't

u can create one model for request with needed fields and best practice is creating new model for response, for sqlalchemy u can return immediately model of db by set ur pydantic model in hints with 
```
class Config:
        orm_mode = True
```
in your pydantic model
